### PR TITLE
Restrict direct HTTP access

### PR DESCRIPTION
**Pull request by baby-gnu**
* .htaccess: limit access to js, css and png files

### CODE REVIEW

**Comment by baby-gnu** [comment]
I don't see how my commit can make the `TransitionTest.php` to fail. Any idea about what's going on?

File: .htaccess [hunk 1]
```diff
@@ -14,6 +14,19 @@
...
 
     RewriteEngine On
+
+    # Deny access
+    RewriteRule "^config(?:.default)?\.php" "-" [NC,F]
+    RewriteRule "^web\.config" "-" [NC,F]
+    RewriteRule "^(?:cli|ChangeLog|LICENSE)" "-" [NC,F]
+
+    # Skip the deny rule for js, css and png
+    RewriteCond "%{REQUEST_URI}" "^/(?:venor|plugins)/.*\.(?:js|css|png)$"
```
Comment (fguillot) [inline .htaccess]: typo here: `venor`

File: .htaccess [hunk 2]
```diff
@@ -14,6 +14,19 @@
...
+    RewriteRule ".?" "-" [S=1]
+
+    # Deny access to any non assets plugin files
+    RewriteRule "^(?:vendor|plugins)/" "-" [NC,F]
```
Comment (fguillot) [inline .htaccess]: Some plugins like the Calendar, Gantt or Chat contains assets. If do `curl http://example.org/plugins/Calendar/Assets/calendar.js` I got a 403.

Comment (baby-gnu) [reply .htaccess]: It's working for me:

```
curl -I https://kanboard.mouvement-democratie-reelle.fr/plugins/Calendar/Assets/calendar.js
HTTP/1.1 200 OK
Set-Cookie: cluster017BAK=R896099146; path=/; expires=Wed, 10-Jan-2018 18:40:54 GMT
Date: Wed, 10 Jan 2018 17:24:27 GMT
Content-Type: application/javascript
Content-Length: 2622
Set-Cookie: cluster017=R3592952301; path=/; expires=Wed, 10-Jan-2018 18:23:26 GMT
Server: Apache
Last-Modified: Sun, 07 Jan 2018 03:05:27 GMT
Accept-Ranges: bytes
Vary: Accept-Encoding
Cache-Control: max-age=900
Expires: Wed, 10 Jan 2018 17:39:28 GMT
X-IPLB-Instance: 17328
```

```
curl -I https://kanboard.mouvement-democratie-reelle.fr/plugins/Calendar/README.md
HTTP/1.1 403 Forbidden
Set-Cookie: cluster017BAK=R896102413; path=/; expires=Wed, 10-Jan-2018 18:40:01 GMT
Date: Wed, 10 Jan 2018 17:24:43 GMT
Content-Type: text/html; charset=iso-8859-1
Set-Cookie: cluster017=R3592952301; path=/; expires=Wed, 10-Jan-2018 18:34:04 GMT
Server: Apache
X-IPLB-Instance: 502
```

File: .htaccess (hunk 1 above)
Comment (baby-gnu) [reply .htaccess]: Thanks I fixed it in last commit.

File: .htaccess (hunk 2 above)
Comment (baby-gnu) [reply .htaccess]: I confess that my OVH hoster had some security activated and rejected the user agent of curl, so I disabled that security to make my tests.

Regards.

File: .htaccess [hunk 3]
```diff
@@ -14,6 +14,19 @@
...
+    RewriteRule "^config(?:.default)?\.php" "-" [NC,F]
+    RewriteRule "^web\.config" "-" [NC,F]
+    RewriteRule "^(?:cli|ChangeLog|LICENSE)" "-" [NC,F]
+
+    # Skip the deny rule for js, css and png
+    RewriteCond "%{REQUEST_URI}" "^/(?:vendor|plugins)/.*\.(?:js|css|png)$"
+    RewriteRule ".?" "-" [S=1]
+
+    # Deny access to any non assets plugin files
+    RewriteRule "^(?:vendor|plugins)/" "-" [NC,F]
```
Comment (fguillot) [inline .htaccess]: Your are still blocking the plugins folder?

Comment (baby-gnu) [reply .htaccess]: Yes, and my curl are working correctly:

```
curl -I https://kanboard.mouvement-democratie-reelle.fr/plugins/Calendar/Assets/calendar.js
HTTP/1.1 200 OK
Set-Cookie: cluster017BAK=R896100235; path=/; expires=Mon, 15-Jan-2018 19:52:56 GMT
Date: Mon, 15 Jan 2018 18:52:38 GMT
Content-Type: application/javascript
Content-Length: 2622
Set-Cookie: cluster017=R3592744302; path=/; expires=Mon, 15-Jan-2018 19:54:50 GMT
Server: Apache
Last-Modified: Sun, 07 Jan 2018 03:05:27 GMT
Accept-Ranges: bytes
Vary: Accept-Encoding
Cache-Control: max-age=0, no-store
Expires: Mon, 15 Jan 2018 19:07:38 GMT
X-IPLB-Instance: 1049
```

```
curl -I https://kanboard.mouvement-democratie-reelle.fr/plugins/Calendar/README.md
HTTP/1.1 403 Forbidden
Set-Cookie: cluster017BAK=R896107858; path=/; expires=Mon, 15-Jan-2018 20:02:48 GMT
Date: Mon, 15 Jan 2018 18:52:33 GMT
Content-Type: text/html; charset=iso-8859-1
Set-Cookie: cluster017=R3592744302; path=/; expires=Mon, 15-Jan-2018 19:57:56 GMT
Server: Apache
X-IPLB-Instance: 17323
```

As I explained before, the curl user-agent was blocked by my hosting (OVH) and I nedded to relax the security for it to work. It always passed in my browser.

Regards.

Comment (fguillot) [reply .htaccess]: It seems to work if Kanboard is installed at the document root, but people could install Kanboard into a subfolder like this: `https://localhost/kanboard/`. In this scenario I get a 403.

Comment (baby-gnu) [reply .htaccess]: OK, I did not try this one…

Comment (fguillot) [reply .htaccess]: Are you planning to update your pull-request?

**Comment by baby-gnu** [comment]
Hello and sorry for the late response.

I do not use Apache except for one instance because of the hoster.

I have little interest in this and less time to test other configurations.

Thanks a lot.

**Pull request closed without merging**